### PR TITLE
Move VEP to Requester Pays

### PR DESCRIPTION
**Pull request by johnc1231**
This PR changes our vep init scripts to pull from the new `hail-us-vep` google bucket, which is requester pays. 

I realized while doing this that my previous PR (#8253) really had nothing to do with supporting this, since that really just sets the configuration for the Google Cloud / Hadoop connector, which is not how we get vep data. I tried to enforce the rules from those command line arguments anyway by rejecting `--vep` flag if they don't specify they're ok with requester pays and manually checking that `hail-us-vep` was in their approved bucket list. But if a user was to specify the init scripts using `gcloud dataproc` and didn't go through `hailctl`, there'd be no catch to check if they were ok with requester pays. Perhaps there is some way we could set environment variables on the dataproc machines based on the `--requester-pays-allow....` flags and use those in the init scripts.

I also expanded `make test-dataproc` to test with a GRCh38 cluster as well, as we use separate scripts to make them and I'm uncomfortable with only testing one.

### CODE REVIEW

**Comment by johnc1231** [comment]
Adding "WIP" tag since I had a test failure from `make test-dataproc`. Seems like I have a mix of reference genomes somewhere.

**Comment by johnc1231** [comment]
Ah, the mix of reference genomes is coming from me running all the "cluster_tests" for GRCh38 cluster as well. One of the tests hardcodes 37

**Comment by johnc1231** [comment]
Ok, all tests passing now. Removing WIP

File: hail/python/hailtop/hailctl/dataproc/start.py [hunk 1]
```diff
@@ -148,7 +148,12 @@ def main(args, pass_through_args):
 
     # add VEP init script
     if args.vep:
+        allowed_buckets_list = args.requester_pays_allow_buckets.split(",") if args.requester_pays_allow_buckets else []
+        if not (args.requester_pays_allow_all or ('hail-us-vep' in allowed_buckets_list)):
+            raise RuntimeError("Need to enable requester pays on bucket 'hail-us-vep' to use vep. See --requester-pays-allow-all and --requester-pays-allow-buckets \
+                WARNING: This is likely prohibitively expensive if you are outside the US. Contact us if you need VEP outside of the US.")
         conf.extend_flag('initialization-actions', [deploy_metadata[f'vep-{args.vep}.sh']])
+        sys.stderr.write("WARNING: Running VEP outside of the US is VERY expensive. Please contact us for support.")
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/dataproc/start.py]: "Please contact the Hail team on https://discuss.hail.is for support."

Comment (tpoterba) [reply hail/python/hailtop/hailctl/dataproc/start.py]: (and in the message above)

**Pull request merged**